### PR TITLE
Out of range in ipfs add

### PR DESCRIPTION
**Pull request by MichaelMure**
I triggered an out of range in coreunix/add.go when adding a file through the API (I accidentally made a POST query without an actual file content).

Fixed it and documented/cleaned the code in the process.

### CODE REVIEW

File: core/coreunix/add.go [hunk 1]
```diff
@@ -181,7 +178,7 @@ func (adder *Adder) Finalize() (*dag.Node, error) {
 	}
 
 	var name string
-	if !adder.Wrap {
+	if !adder.Wrap && len(root.Links) == 1 {
```
Comment (MichaelMure) [inline core/coreunix/add.go]: This is the fix.

**Comment by whyrusleeping** [comment]
Good catch, Do you think you could add a sharness test for this somewhere?

maybe in `tests/sharness/t0235...`?

**Comment by MichaelMure** [comment]
Will do

**Comment by whyrusleeping** [comment]
@MichaelMure the codepath youre toching in the PR has changed. Can you rerun your repro and see if this issue still occurs on master?

**Comment by whyrusleeping** [comment]
I don't think the bugfix here is relevant any longer, but the documentation would be nice.

**Comment by whyrusleeping** [comment]
Closing due to inactivity, please reopen as needed.

**Pull request closed without merging**